app/views/layouts/_top_bar.html.erb

```diff
@@ -41,63 +41,85 @@
 
       <div class="top-bar__menu" id="top-bar__menu">
         <button type="button" class="top-bar__menu__trigger" id="navigation-butt" aria-label="Navigation menu">
-          <% if user_signed_in? %>
-            <span class="crayons-avatar crayons-avatar--l"><img class="crayons-avatar__image" alt="" id="nav-profile-image" /></span>
-          <% else %>
-            <%= inline_svg_tag("hamburger.svg", aria: true, class: "crayons-icon", title: "Navigation menu") %>
-          <% end %>
+          <span class="crayons-avatar crayons-avatar--l"><img class="crayons-avatar__image" alt="" id="nav-profile-image" /></span>
+        </button>
+        <%= render "layouts/nav_menu" %>
+      </div>
+    </div>
+
+    <div id="top-bar-logged-out" class="top-bar__links <%= user_signed_in? ? "hidden" : "" %>">
+      <a href="/enter" class="crayons-btn crayons-btn--ghost-brand hidden mr-2 whitespace-nowrap s:block">
+        Log in
+      </a>
+      <a href="/enter" class="crayons-btn hidden mr-2 whitespace-nowrap s:block">
+        Create account
+      </a>
+
+      <div class="top-bar__menu s:hidden" id="top-bar__menu">
+        <a id="moderation-link" class="hidden top-bar__link trusted-visible-block" aria-label="Moderation" href="<%= mod_path %>">
+          <%= inline_svg_tag("mod.svg", aria: true, class: "crayons-icon", title: "Moderation") %>
+        </a>
+
+        <a href="/connect" id="connect-link" class="top-bar__link hidden" aria-label="Connect">
+          <%= inline_svg_tag("connect.svg", aria: true, class: "crayons-icon", title: "Connect") %>
+          <span class="crayons-indicator crayons-indicator--accent hidden" id="connect-number"></span>
+        </a>
+        <a href="/notifications" id="notifications-link" class="top-bar__link hidden" aria-label="Notifications">
+          <%= inline_svg_tag("bell.svg", aria: true, class: "crayons-icon", title: "Notifications") %>
+          <span class="crayons-indicator crayons-indicator--critical hidden" id="notifications-number"></span>
+        </a>
+        <button type="button" class="top-bar__menu__trigger" id="navigation-butt" aria-label="Navigation menu">
+          <%= inline_svg_tag("hamburger.svg", aria: true, class: "crayons-icon", title: "Navigation menu") %>
         </button>
         <%= render "layouts/nav_menu" %>
       </div>
     </div>
   </div>
-  <% if user_signed_in? %>
-    <script async>
-      // Here we have some scripts we want to get working on before
-      // waiting for the page to stream in.
+  <script async>
```

Comment:
I don't think this needs to be changed.... There may have been some confusion about this internally eh?

Reply:
It was indeed a confusion. The bug we saw before was unrelated to this logic. I'm working on getting the last few specs to pass and also taking it back to the previous logic 👍